frappe/auth.py

```diff
@@ -209,7 +209,16 @@ def authenticate(self, user = None, pwd = None):
 			self.fail(_('Incomplete login details'), user=user)
 
 		# Ignore password check if tmp_id is set, 2FA takes care of authentication.
-		validate_password = not bool(frappe.form_dict.get('tmp_id'))
+		tmp_id = bool(frappe.form_dict.get('tmp_id'))
+
+		# Check if 2FA is enabled or not
+		enabled = int(frappe.db.get_value('System Settings', None, 'enable_two_factor_auth') or 0)
+
+		validate_password = True
+		
+		if tmp_id and enabled:
+			validate_password = False
+		
```

Comment:
There's nothing technically wrong with the proposed changes but I believe it could be cleaner so here's a nitpick

```suggestion
		tmp_id = bool(frappe.form_dict.tmp_id)
		enabled = cint(frappe.db.get_single_value('System Settings', 'enable_two_factor_auth'))
		# TODO: Add comment about what the next line does
		validate_password = not (tmp_id and enabled)
```